spec/models/nested_facet_spec.rb

```diff
@@ -131,5 +138,46 @@
         )
       end
     end
+
+    context "when there is a selection" do
```

Comment:
Thanks for the detailed explanation in the commit message 👍 

This is a good unit test. It feels like we may also want an integration test (perhaps somewhere like [finders_controller_spec](https://github.com/alphagov/finder-frontend/blob/main/spec/controllers/finders_controller_spec.rb)), to properly verify that a change in URL parameter is reflected in which options have `selected="selected"` set?